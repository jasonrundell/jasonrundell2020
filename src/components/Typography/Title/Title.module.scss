@import '../../../../src/styles/main.scss';

.h1 {
  font-size: $h1Size;
  font-weight: $h1Weight;
  line-height: $h1Size;
  margin-top: $h1MarginTop;
  margin-bottom: $h1MarginBottom;
}

.h2 {
  font-size: $h2Size;
  font-weight: $h2Weight;
  line-height: $h2Size;
  margin-top: $h2MarginTop;
  margin-bottom: $h2MarginBottom;
}

.h3 {
  font-size: $h3Size;
  font-weight: $h3Weight;
  line-height: $h3Size;
  margin-top: $h3MarginTop;
  margin-bottom: $h3MarginBottom;
}

.h4 {
  font-size: $h4Size;
  font-weight: $h4Weight;
  line-height: $h4Size;
  margin-top: $h4MarginTop;
  margin-bottom: $h4MarginBottom;
}
