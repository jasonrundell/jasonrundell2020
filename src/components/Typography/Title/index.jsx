import React from 'react'
import PropTypes from 'prop-types'

const Title = ({ level, children }) => {
  const Component = `h${level}`
  return <Component>{children}</Component>
}

Title.defaultProps = {
  level: 1,
}

Title.propTypes = {
  level: PropTypes.oneOf([1, 2, 3, 4]),
  children: PropTypes.any.isRequired,
}

export default Title
