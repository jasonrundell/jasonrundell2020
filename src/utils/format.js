/**
 *  Letter case conversion functions
 *  [Letter case](https://en.wikipedia.org/wiki/Letter_case#Special_case_styles)
 *
 *  Converts only letter case-separated or hyphen-delimited strings
 */

/**
 *  Converts hyphen-delimited `string` to camel case
 *
 *  @example
 *  'foo-bar' => 'fooBar'
 *  'foo--bar' => 'fooBar'
 *
 *  @param {string} string
 */
function toCamelCase(string) {
  return string.replace(/(-+)([a-z])/g, ($0, $1, $2) => $2.toUpperCase())
}

/**
 *  Converts hyphen-delimited `string` to pascal case (aka upper camel case)
 *
 *  @example
 *  'foo-bar' => 'FooBar'
 *  'foo--bar' => 'FooBar'
 *
 *  @param {string} string
 */
function toPascalCase(string) {
  return toCamelCase(string.replace(/^[a-z]/, $0 => $0.toUpperCase()))
}

export { toCamelCase, toPascalCase }
