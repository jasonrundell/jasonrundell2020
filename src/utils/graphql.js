export function edgesToObject(edgesData) {
  const nodes = edgesData.map(item => {
    return item.node
  })
  const finalObject = nodes.map(item => {
    return item
  })

  return finalObject
}
