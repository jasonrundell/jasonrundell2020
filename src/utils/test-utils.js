export function catchWarnings() {
  const originalConsoleError = global.console.error
  beforeEach(() => {
    /**
     * I Want to know if values passed to props fail the type check
     * */
    global.console.error = (...args) => {
      const propTypeFailures = [/Warning:/] // e.g. [/Invalid prop/, /Failed prop type/]

      if (propTypeFailures.some(p => p.test(args[0]))) {
        throw new Error(args[0])
      }

      originalConsoleError(...args)
    }
  })
}

/**
 * Mix an match help from:
 * https://daveceddia.com/testing-for-proptypes-errors/
 * https://www.jackfranklin.co.uk/blog/failing-tests-on-react-proptypes/
 */
