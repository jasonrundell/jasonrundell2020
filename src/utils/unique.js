const onlyUnique = (value, index, self) => {
  return self.indexOf(value) === index
}

const generatePassword = ({
  passwordLength = 16,
  charset = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_-=~`',
}) => {
  let retVal = ''
  for (var i = 0, n = charset.length; i < passwordLength; ++i) {
    retVal += charset.charAt(Math.floor(Math.random() * n))
  }
  return retVal
}

export { onlyUnique, generatePassword }
