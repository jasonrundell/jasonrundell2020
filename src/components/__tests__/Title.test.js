import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import renderer from 'react-test-renderer'
import { catchWarnings } from '../../utils/test-utils'

import Title from '../Typography/Title'

catchWarnings()

let container = null
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

describe('Title component', () => {
  it('Title renders correctly', () => {
    const tree = renderer
      .create(
        <div>
          <Title>Title component</Title>
          <Title level={1}>Title component level 1</Title>
          <Title level={2}>Title component level 2</Title>
          <Title level={3}>Title component level 3</Title>
          <Title level={4}>Title component level 4</Title>
        </div>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Title has required children prop', () => {
    act(() => {
      render(<Title>Title component</Title>, container)
    })
    expect(container.textContent).toBe('Title component')
  })
})
