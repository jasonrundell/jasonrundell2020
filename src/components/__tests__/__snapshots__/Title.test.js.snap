// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component Title renders correctly 1`] = `
<div>
  <h1
    className="h1"
  >
    Title component
  </h1>
  <h1
    className="h1"
  >
    Title component level 1
  </h1>
  <h2
    className="h2"
  >
    Title component level 2
  </h2>
  <h3
    className="h3"
  >
    Title component level 3
  </h3>
  <h4
    className="h4"
  >
    Title component level 4
  </h4>
</div>
`;
